### PR TITLE
Repeat call view on "virtual state has been invalidated"

### PR DESCRIPTION
**Pull request by Juliusan**
A quick update to repeat call view in `counterEquals` function. In the spirit of https://github.com/iotaledger/wasp/pull/248.

### CODE REVIEW

**Review by kape1395 — APPROVED** [review]

File: tools/cluster/tests/util.go [hunk 1]
```diff
@@ -239,11 +240,27 @@ func waitTrue(timeout time.Duration, fun func() bool) bool {
 	}
 }
 
+func repeatIfInvalidated(fun func() (dict.Dict, error), deadline time.Time) (dict.Dict, error) {
+	result, err := fun()
+	if err != nil {
+		errStr := fmt.Sprintf("%v", err)
+		if strings.Contains(errStr, "virtual state has been invalidated") {
```
Comment (jorgemmsilva) [inline tools/cluster/tests/util.go:247]: not a fan of checking the error by string matching 😕 
could we create a special kind of error and use `errors.Is`? 
https://pkg.go.dev/errors#Is

File: tools/cluster/tests/util.go [hunk 2]
```diff
@@ -239,11 +240,27 @@ func waitTrue(timeout time.Duration, fun func() bool) bool {
...
+			if time.Now().Before(deadline) {
+				return repeatIfInvalidated(fun, deadline)
```
Comment (jorgemmsilva) [inline tools/cluster/tests/util.go:249]: should we add some waiting time here before re-trying?

File: tools/cluster/tests/util.go (hunk 1 above)
Comment (Juliusan) [reply tools/cluster/tests/util.go:247]: Not a fan either. That, I believe, should depend on error code returned from the server. It is done more elegantly on the server side, but do we want to do it here or there? That's a discussion I wanted.

File: tools/cluster/tests/util.go (hunk 2 above)
Comment (Juliusan) [reply tools/cluster/tests/util.go:249]: Waiting is done on `RepeatOnceIfUnlucky`. I hoped to do without it here... :-)

File: tools/cluster/tests/util.go (hunk 1 above)
Comment (jorgemmsilva) [reply tools/cluster/tests/util.go:247]: oh okay, I didn't realize this was checking on a http result

**Review by lunfardo314 — APPROVED** [review]

**Pull request merged**